File: MH/DAY2/11726.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
let input = fs.readFileSync(filePath).toString().trim();
let x = +input;
let arr = [0, 1, 2];

for (let i = 3; i <= x; i++) {
  arr[i] = (arr[i - 1] + arr[i - 2]) % 10007;
}
console.log(arr[x]);

File: GH/day1/11726.py
n = int(input())

dp = [0] * 1001 # 1001이 아닌 (n+1)을 곱하면 런타임 에러가 나온다. 이유는 모름

dp[1] = 1
dp[2] = 2

for i in range(3, n+1):
  dp[i] = dp[i-1] + dp[i-2]

print(dp[n]%10007)

